file-ingestion-test/src/main/java/com/hazelcast/jet/tests/file/ingestion/FileIngestionTest.java

```diff
@@ -0,0 +1,252 @@
+import org.apache.hadoop.hdfs.DistributedFileSystem;
+import org.apache.hadoop.mapred.JobConf;
+import org.apache.hadoop.mapred.TextOutputFormat;
+import software.amazon.awssdk.auth.credentials.AwsBasicCredentials;
+import software.amazon.awssdk.auth.credentials.StaticCredentialsProvider;
+import software.amazon.awssdk.core.sync.RequestBody;
+import software.amazon.awssdk.http.apache.ApacheHttpClient;
+import software.amazon.awssdk.regions.Region;
+import software.amazon.awssdk.services.s3.S3Client;
+import software.amazon.awssdk.services.s3.model.PutObjectRequest;
+
+import java.net.URI;
+import java.nio.file.Paths;
+import java.time.Duration;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.IntStream;
+
+import static com.hazelcast.function.FunctionEx.identity;
+import static com.hazelcast.jet.aggregate.AggregateOperations.counting;
+import static com.hazelcast.jet.tests.common.Util.sleepSeconds;
+import static java.util.stream.Collectors.toList;
+
+public class FileIngestionTest extends AbstractSoakTest {
+
+
+    private static final String DEFAULT_DIRECTORY = "/tmp/" + FileIngestionTest.class.getSimpleName();
+    private static final String DEFAULT_HDFS_URI = "hdfs://localhost:8020";
+    private static final String DEFAULT_BUCKET_NAME = "jet-soak-tests-bucket";
+    private static final int S3_CLIENT_CONNECTION_TIMEOUT_SECONDS = 10;
+    private static final int S3_CLIENT_SOCKET_TIMEOUT_MINUTES = 5;
+    private static final int DEFAULT_SLEEP_SECONDS = 4;
+    private static final int ITEM_COUNT = 1000;
+
+    private String localDirectory;
+    private String hdfsUri;
+    private String hdfsPath;
+    private String bucketName;
+    private String s3Directory;
+    private String accessKey;
+    private String secretKey;
+    private int sleepSeconds;
+
+    public static void main(String[] args) throws Exception {
+        new FileIngestionTest().run(args);
+    }
+
+    @Override
+    protected void init(JetInstance client) throws Exception {
+        localDirectory = property("localDirectory", DEFAULT_DIRECTORY);
+
+        hdfsUri = property("hdfsUri", DEFAULT_HDFS_URI);
+        hdfsPath = hdfsUri + "/" + getClass().getSimpleName();
+
+        bucketName = property("bucketName", DEFAULT_BUCKET_NAME);
+        s3Directory = getClass().getSimpleName() + "/";
+        accessKey = property("accessKey", "");
+        secretKey = property("secretKey", "");
+
+        sleepSeconds = propertyInt("sleepSecondsBetweenJobs", DEFAULT_SLEEP_SECONDS);
+
+        createSourceFiles(client);
+    }
+
+    @Override
+    protected void test(JetInstance client, String name) {
+        long begin = System.currentTimeMillis();
+        int jobNumber = 0;
+        JobType[] jobTypes = JobType.values();
+        Job[] jobs = new Job[jobTypes.length];
+        while ((System.currentTimeMillis() - begin) < durationInMillis) {
+            for (int i = 0; i < jobTypes.length; i++) {
```

Comment:
Would not be better to have this part in different threads - 1 thread for every `JobType`. Currently if any of job fails it will fail whole test. If any of job gets stuck it will stop processing all of them etc.